src/main/java/com/blackwaterpragmatic/webservices/resource/UserResource.java

```diff
@@ -30,10 +32,13 @@ public UserResource(
 
 	@GET
 	@Produces("application/json;charset=utf-8")
-	public Response getUsers() {
-		final List<User> users = userService.listUsers();
+	public Response getUsers(@QueryParam("name") final String name) {
+		final List<User> users = userService.listUsers(name).stream()
```

Comment:
I elected to sort the collection here instead of the service/mapper because I did not want to modify those classes since other callers might not want this behaviour.